Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm

```diff
@@ -81,37 +81,42 @@
     return byteCast<uint8_t>(m_token.span());
 }
 
-char* SandboxExtensionImpl::sandboxExtensionForType(const char* path, SandboxExtension::Type type, std::optional<audit_token_t> auditToken, OptionSet<SandboxExtension::Flags> flags)
-{
-    uint32_t extensionFlags = 0;
-    if (flags & SandboxExtension::Flags::NoReport)
-        extensionFlags |= SANDBOX_EXTENSION_NO_REPORT;
-    if (flags & SandboxExtension::Flags::DoNotCanonicalize)
-        extensionFlags |= SANDBOX_EXTENSION_CANONICAL;
-
-    switch (type) {
-    case SandboxExtension::Type::ReadOnly:
-        return sandbox_extension_issue_file(APP_SANDBOX_READ, path, extensionFlags);
-    case SandboxExtension::Type::ReadWrite:
-        return sandbox_extension_issue_file(APP_SANDBOX_READ_WRITE, path, extensionFlags);
-    case SandboxExtension::Type::Mach:
-        if (!auditToken)
-            return sandbox_extension_issue_mach("com.apple.webkit.extension.mach", path, extensionFlags);
-        return sandbox_extension_issue_mach_to_process("com.apple.webkit.extension.mach", path, extensionFlags, *auditToken);
-    case SandboxExtension::Type::IOKit:
-        if (!auditToken)
-            return sandbox_extension_issue_iokit_registry_entry_class("com.apple.webkit.extension.iokit", path, extensionFlags);
-        return sandbox_extension_issue_iokit_registry_entry_class_to_process("com.apple.webkit.extension.iokit", path, extensionFlags, *auditToken);
-    case SandboxExtension::Type::Generic:
-        return sandbox_extension_issue_generic(path, extensionFlags);
-    case SandboxExtension::Type::ReadByProcess:
-        if (!auditToken)
-            return nullptr;
+CString SandboxExtensionImpl::sandboxExtensionForType(const char* path, SandboxExtension::Type type, std::optional<audit_token_t> auditToken, OptionSet<SandboxExtension::Flags> flags)
+{
+    auto sandboxExtensionString = [&] {
```

Comment:
The `auto*` here is a `char*` that needs freeing. This is too error prone, we want a smart pointer *right away*, like I originally suggested.

Reply:
Ah, yes, good point, fixed in latest patch.

Thanks for reviewing!